sdk/include/ndk/amd64/ketypes.h

```diff
@@ -276,13 +276,17 @@ typedef enum
 //
 #define KSEG0_BASE 0xfffff80000000000ULL
 
+#define NMI_STACK_SIZE 0x2000
+#define ISR_STACK_SIZE 0x6000
+
 //
 // Synchronization-level IRQL
 //
-#define SYNCH_LEVEL 12
-
-#define NMI_STACK_SIZE 0x2000
-#define ISR_STACK_SIZE 0x6000
+#ifndef CONFIG_SMP
```

Comment:
I never thought this made a whole lot of sense the way it's done for x86. `CONFIG_SMP` has no meaning outside of the kernel, in most cases it won't be defined, therefore you get the UP version, no matter what type of system you're running on.
Perhaps something like `#define SYNCH_LEVEL_UP ...` / `#define SYNCH_LEVEL_MP ...` would be more useful?

Reply:
NDK contains a lot of stuff that should be private to the kernel. But this is what we do and we need this definition somewhere, I guess.

Reply:
Those definitions may also be used in the HAL.